core/interfaces/branch.py

```diff
@@ -502,18 +502,23 @@ def run(self, edit):
 
     def run_async(self):
         self.interface = ui.get_interface(self.view.id())
-        current_branch_name = self.get_current_branch_name()
 
+        branches = self.get_selected_branches()
+        branches_strings = self.create_branches_strs(branches)
+        self.merge(branches_strings) if not len(branches_strings) == 0 else None
```

Comment:
Once [`BranchesMixin.merge()`][1] is fixed (*required*) this should be simplified to `self.merge(branches_strings)` without the ternary non-empty confirmation.

[1]: https://github.com/divmain/GitSavvy/blob/e9bd976a8e20b4bbec29602468557d2dcc89b245/core/git_mixins/branches.py#L79-L84